crates/bevy_ecs/src/component.rs

```diff
@@ -12,26 +12,85 @@ use std::{
     any::{Any, TypeId},
 };
 
-/// A component is data associated with an [`Entity`](crate::entity::Entity). Each entity can have
-/// multiple different types of components, but only one of them per type.
+// TODO: The content in the section “Implementing the trait for foreign types” may be more appropriate
+// for the Bevy book. However, until the change gets published, it's better to keep it there. Once the
+// book includes usage of the pattern, the section can be replaced with a link to the appropriate
+// section of the book.
+
+/// A data type that can be associated to an [entity].
+///
+/// `Component` is a [derivable trait]:
+/// this means that a data type can implement it by simply prepending a `#[derive(Component)]` attribute
+/// to it. It is necessary though that it satisfies the `Send + Sync + 'static` bounds.
+///
+/// In the context of a [`World`], a component instance must belong to a single entity.
+///
+/// [entity]: crate::entity
+/// [derivable trait]: https://doc.rust-lang.org/book/appendix-03-derivable-traits.html
+/// [`World`]: crate::world::World
+///
+/// # Examples
 ///
-/// Any type that is `Send + Sync + 'static` can implement `Component` using `#[derive(Component)]`.
+/// Components can take many forms: they are usually structs, but can also be of every other kind of data type, like enums or zero sized types. The following examples show how components are laid out in code.
 ///
-/// In order to use foreign types as components, wrap them using a newtype pattern.
 /// ```
 /// # use bevy_ecs::component::Component;
+/// # struct Color;
+/// #
+/// // A component can contain data...
+/// #[derive(Component)]
+/// struct LicensePlate(String);
+///
+/// // ... but it can also be a zero-sized marker.
+/// #[derive(Component)]
+/// struct Car;
+///
+/// // Most commonly, components are structs with a small number of fields.
+/// #[derive(Component)]
+/// struct VehicleProperties {
+///     wheel_number: usize,
+///     color: Color,
+/// }
+/// ```
```

Comment:
IMO this should demo enum components too. It wasn't intuitive to me at first that this would work and it's a great pattern.

Reply:
Yep, that's more plausible.